src/lib/zkstring.ts

```diff
@@ -0,0 +1,85 @@
+import { Bool, Circuit, Field } from '../snarky';
+import { CircuitValue, prop } from './circuit_value';
+
+export class Character extends CircuitValue {
+  @prop value: Field;
+
+  constructor(value: Field) {
+    super();
+
+    Character.validateInput(value);
+    this.value = value;
+  }
+
+  equals(char: Character): Bool {
+    return this.value.equals(char.value);
+  }
+
+  toField(): Field {
+    return this.value;
+  }
+
+  toString(): string {
+    const charCode = Number(this.value.toString());
+    return String.fromCharCode(charCode);
+  }
+
+  static fromString(str: string) {
+    const char = Field(str.charCodeAt(0));
+    return new Character(char);
+  }
+
+  static validateInput(value: Field): Boolean {
+    if (true) {
+      return true;
```

Comment:
Here could add things like is within certain range e.g. utf-8 1-byte characters, or any utf-8 character, or even this could be an abstract class and different character sets would need to implement their own validations?